models/ldndc/R/met2model.LDNDC.R

```diff
@@ -0,0 +1,247 @@
+      
+      # On above the starting and ending indexes has been determined.
+      # Here those are just collapsed.
+      ind <- ind_s:ind_e
+      
+        
+      
+      ## Info for climate file that model is using ##
+      # Latitude and longitude
+      lat <- ncdf4::ncvar_get(nc, "latitude")
+      lon <- ncdf4::ncvar_get(nc, "longitude")
+      
+      
+      # Average air temperature
+      Tair <-ncdf4::ncvar_get(nc, "air_temperature")[ind]  ## in Kelvin
+      tavg <- udunits2::ud.convert(Tair, "K", "degC")
+      
+      # Wind speed
+      wind <- try(ncdf4::ncvar_get(nc, "wind_speed"))[ind]
+      
+      # Precipation
+      prec <- ncdf4::ncvar_get(nc, "precipitation_flux")[ind]
+      
+      # Global radiation
+      grad <- ncdf4::ncvar_get(nc, "surface_downwelling_shortwave_flux_in_air")[ind]  ## in W/m2
+      
+      # Vapor Pressure Deficit Pa ---> kPa, need to ne divided by 1000
+      VPD <- try(ncdf4::ncvar_get(nc, "water_vapor_saturation_deficit"))[ind]  ## in Pa
+      
+      # Relative humidity (%)
+      rhum <- ncdf4::ncvar_get(nc, "relative_humidity")[ind]
+      
+      # Air pressure Pa ---> mbar, needs to divided by 100
+      press <- ncdf4::ncvar_get(nc, "air_pressure")[ind]
+      
+      
+      # Close connection after all necessary values have been fetch
+      ncdf4::nc_close(nc)
+      
+
+      # Gather the vectors to dataframe
+      data <- as.data.frame(do.call("cbind", list(y = y, d = d, s = subd, prec = prec*86400/tstep, tavg = tavg,
+                                    grad = grad, vpd = VPD/1000, wind = wind, press = press/100)))
+      
+      # Write prefix before the actual data
+      if(year == start_year){
+        
+        # General information before the daily/subdaily climate data
+        prefix_global <- paste0('\t time = "', start_date, '/', tstep, '"')
+        prefix_climate <- paste0('\t id = "', 0, '"')
+        prefix_latitude <- paste0('\t latitude = "', lat, '"')
+        prefix_longitude <- paste0('\t longitude = "', lon, '"')
+        
+        data_prefix <- paste(#"%global", prefix_global,
+                             "%climate", prefix_climate,
+                             "%attributes", prefix_latitude, prefix_longitude,
+                             "%data \n", sep = "\n")
+        
+        # Write prefix information before the data
+        cat(data_prefix, file = file.path(outfolder, out.file))
+        
+        # For the first year, keep col.names as TRUE
+        data.table::fwrite(x = data, file = file.path(outfolder, out.file),
+                           sep = "\t", col.names = T, append = T)
+        
+      }else{
+        # For the other year, col.names are FALSE
+        data.table::fwrite(x = data, file = file.path(outfolder, out.file),
+                           sep = "\t", col.names = F, append = T)
+      }
+      
+      
+      
+    } else{
+      PEcAn.logger::logger.info(paste(old.file, "file does not exist"))
```

Comment:
Is it fine for the model run that the other years do not exist? If not, consider failing here, i.e. using `logger.severe` instead of `logger.info`